sklearn/pipeline.py

```diff
@@ -185,6 +190,17 @@ def named_steps(self):
     def _final_estimator(self):
         return self.steps[-1][1]
 
+    def _log_message(self, step_idx):
+        if not self.verbose:
+            return
+        if step_idx < 0:
+            step_idx = len(self.steps) + step_idx
```

Comment:
Unless I'm mistaken, `step_idx` will never be negative, which means there is no need to support index bounds wrapping.